smile_id_core/web_api.py

```diff
@@ -106,6 +95,50 @@ def verify_document(
         if last_name:
             payload["last_name"] = last_name
 
+        return payload
+
+    def verify_document_async(
+        self,
+        **data
+    ):
+        """
+            Performs a document verification request in async way
+
+            :param country:
+            :param id_type:
+            :param id_number:
+            :param first_name: Optional; required for some ID types, e.g. DRIVERS_LICENSE, PASSPORT
+            :param last_name: Optional; required for some ID types
+            :param dob: Optional; required for some ID types. Can be a date
+            :param job_id: Optional; Will be passed to SmileID as partner parameters
+            :param user_id: Optional; Will be passed to SmileID as partner parameters
+
+            :return: dict
+        """
+        payload = self._prepare_document_verification_payload(**data)
+        response = self._make_request_async("POST", self.Urls.VERIFY_DOCUMENT_ASYNC, payload)
+        return response
+
+    def verify_document(
+        self,
+        **data
+    ):
+        """
+        Performs a document verification request with an immediate response.
+
+        :param country:
```

Comment:
If these  parameters are actually used, maybe add them as parameters to the function signature? Otherwise it's just confusing

Reply:
+1

Reply:
Hello, thanks for valuable fixes. I would like to ask about this part. You mean docstring might be confusing? So I guess you mean that moving them to the another function(_prepare_document_verification_payload) where they are used?

Reply:
Yes, the idea is that the docstring should match the arguments.

To be honest, it seems somewhat difficult to carry it out nicely with parameter forwarding. 
Converting this `**data` to kwargs and describing them in both methods (in addition to the `_prepare_document_verification_payload` one) seems to be quite a bit of copy paste.

Moving the descriptions to `_prepare_document_verification_payload` is not optimal because it doesn't provide the info in the appropriate place, but I don't see a better alternative with this approach.

Reply:
okay, I understand. Yeah I moved the params docstring to the _prepare_document_verification_payload and added a note in original method to mention how should **data created. I have fix them and pushing it

Reply:
please also take a look at this comment - https://github.com/ykshatroff/smile-identity-core-python/pull/3#discussion_r636840667